app/src/main/scala/org/alephium/explorer/Explorer.scala

```diff
@@ -0,0 +1,280 @@
+      dc: DatabaseConfig[PostgresProfile],
+      blockFlowClient: BlockFlowClient,
+      blockCache: BlockCache,
+      transactionCache: TransactionCache,
+      groupSetting: GroupSetting): Future[AkkaHttpServer] = {
+    val routes =
+      AppServer.routes()
+
+    val httpServer =
+      Http()
+        .newServerAt(host, port)
+        .bindFlow(routes)
+
+    //routes are required by test-cases.
+    httpServer map { server =>
+      AkkaHttpServer(
+        server      = server,
+        routes      = routes,
+        actorSystem = system
+      )
+    }
+  }
+  // scalastyle:on
+
+  /** Fetch network peers */
+  def getPeers(networkId: NetworkId, directCliqueAccess: Boolean, blockFlowUri: Uri)(
+      implicit ec: ExecutionContext,
+      blockFlowClient: BlockFlowClient): Future[Seq[Uri]] =
+    blockFlowClient
+      .fetchChainParams()
+      .flatMap { chainParams =>
+        val validationResult =
+          validateChainParams(
+            networkId = networkId,
+            response  = chainParams
+          )
+
+        validationResult match {
+          case Failure(exception) =>
+            Future.failed(exception)
+
+          case Success(_) =>
+            getBlockFlowPeers(
+              directCliqueAccess = directCliqueAccess,
+              blockFlowUri       = blockFlowUri
+            )
+        }
+      }
+
+  /**
+    * Initialise the database from the config file.
+    *
+    * TODO: Make config a part of `ExplorerConfig`
+    *
+    * @param readOnly If true tried to validate connection else initialises schema.
+    * @param path     Path of database config in `application.conf` file.
+    * @param ec       Application Level execution (Not used for DB calls)
+    *
+    * @return Slick config provides Ability to run queries.
+    */
+  def initialiseDatabase(readOnly: Boolean, path: String, config: Config)(
+      implicit ec: ExecutionContext): Future[DatabaseConfig[PostgresProfile]] =
+    managed(DatabaseConfig.forConfig[PostgresProfile](path, config)) { implicit dc =>
+      if (readOnly) {
+        HealthCheckDao.healthCheck().mapSyncToVal(dc)
+      } else {
+        DBInitializer.initialize().mapSyncToVal(dc)
+      }
+    }
+
+  /** Converts `PeerAddress` to `Uri` */
+  def urisFromPeers(peers: Seq[PeerAddress]): Seq[Uri] =
+    peers.map { peer =>
+      s"http://${peer.address.getHostAddress}:${peer.restPort}"
+    }
+
+  def getBlockFlowPeers(directCliqueAccess: Boolean, blockFlowUri: Uri)(
+      implicit ec: ExecutionContext,
+      blockFlowClient: BlockFlowClient): Future[Seq[Uri]] =
+    if (directCliqueAccess) {
+      blockFlowClient.fetchSelfClique() flatMap {
+        case Right(selfClique) =>
+          //TODO: What happens when peers are empty?
+          //      Should Application even be allowed to start if there were no peers?
+          val peers = urisFromPeers(selfClique.nodes.toSeq)
+          logger.debug(s"Syncing with clique peers: $peers")
+          Future.successful(peers)
```

Comment:
@tdroxler is this TODO right? Should we fail boot-up if peers return empty or is some empty check required here?

Reply:
yes we should fail, if peers are empty something is very wrong, so better to stop.